src/eventHandling.js

```diff
@@ -40,33 +40,67 @@ function sayCorrectOrIncorrect(span, correct) {
 }
 
 function handleSpace() {
-  // check the text has even been loaded
-  if (comparisonWord !== null) {
+  // checks the text has been loaded, and the game has not finished
+  if (comparisonWord !== null && gameOver == false) {
     const inputTextbox = document.querySelector('#input-textbox')
 
-    currentWordSpan.classList.remove("current-word")
-
+    if (currentWordSpan) {
+      currentWordSpan.classList.remove("current-word")
+    }
     const correct = comparisonWord === inputTextbox.value
 
     inputTextbox.value = ''
 
-    if (correct) {
+    if (correct && currentWordSpan) {
       currentWordSpan.classList.add("correct")
-    } else {
+      totalCorrectWords++
+    } else if (currentWordSpan) {
       currentWordSpan.classList.add("incorrect")
+      totalIncorrectWords++
+    }
+
+    if (currentWordSpan) {
+      cleanupTemporaryScreenReaderText(currentWordSpan)
+
+      // get the next word
+      currentWordSpan = wordSpanArray.shift()
+      comparisonWord = comparisonWordsArray.shift()
     }
 
-    cleanupTemporaryScreenReaderText(currentWordSpan)
+    if (currentWordSpan) {
+      currentWordSpan.classList.add("current-word")
+
+      // say what happened with the last word
+      sayCorrectOrIncorrect(currentWordSpan, correct)
+
+      currentWordSpan.focus()
+    }
+  }
+}
 
-    // get the next word
-    currentWordSpan = wordSpanArray.shift()
-    comparisonWord = comparisonWordsArray.shift()
+function startAndStopTimer() {
```

Comment:
better in timer.js as mentioned previously i reckon

Reply:
done